src/utils/observeScrollArea.ts

```diff
@@ -2,17 +2,20 @@ const resizeOptions: ResizeObserverOptions = { box: 'border-box' };
 const mutationObservers: Map<HTMLElement, MutationObserver> = new Map();
 const resizeListeners: Map<HTMLElement, Array<() => void>> = new Map();
 
-const resizeObserver = new ResizeObserver(entries => {
-  const parents = new Set<Element>();
-  for (let i = 0; i < entries.length; i++) {
-    const parent = entries[i].target.parentElement;
-    if (parent && !parents.has(parent)) {
-      parents.add(parent);
-      const listeners = resizeListeners.get(parent) || [];
-      for (let i = 0; i < listeners.length; i++) listeners[i]();
-    }
-  }
-});
+const resizeObserver =
+  typeof window !== 'undefined'
+    ? new ResizeObserver(entries => {
```

Comment:
I would propose to either make a getter or to just check globally for what's called 😅 The latter is probably fine for now

```suggestion
const resizeObserver =
  typeof ResizeObserver !== 'undefined'
    ? new ResizeObserver(entries => {
```